addons/common/fnc_canAddItem.sqf

```diff
@@ -100,27 +100,36 @@ if (
     _checkUniform
     && {TYPE_UNIFORM in _allowedSlots}
     && {
-        // each time subtract whole number of items which can be put in container
-        _count = _count - floor (getContainerMaxLoad uniform _unit * (1 - loadUniform _unit) / _mass);
-        _count <= 0
+        _mass == 0
+        || {
+            // each time subtract whole number of items which can be put in container
+            _count = _count - floor (getContainerMaxLoad uniform _unit * (1 - loadUniform _unit) / _mass);
+            _count <= 0
+        }
     }
 ) exitWith {true};
 
 if (
     _checkVest
     && {TYPE_VEST in _allowedSlots}
     && {
-        _count = _count - floor (getContainerMaxLoad vest _unit * (1 - loadVest _unit) / _mass);
-        _count <= 0
+        _mass == 0
+        || {
+            _count = _count - floor (getContainerMaxLoad vest _unit * (1 - loadVest _unit) / _mass);
+            _count <= 0
+        }
     }
 ) exitWith {true};
 
 if (
     _checkBackpack
     && {TYPE_BACKPACK in _allowedSlots}
     && {
-        _count = _count - floor (getContainerMaxLoad backpack _unit * (1 - loadBackpack _unit) / _mass);
-        _count <= 0
+        _mass == 0
+        || {
+            _count = _count - floor (getContainerMaxLoad backpack _unit * (1 - loadBackpack _unit) / _mass);
+            _count <= 0
+        }
```

Comment:
Can't you check this once before checking the 3 containers?

Reply:
no, it can have mass==0 and allowedSlots not with all containers

Reply:
I see.